web-components/src/components/advance-list/AdvanceList.ts

```diff
@@ -0,0 +1,269 @@
+          clickedItem?.getAttribute("aria-disabled") === "true";
+        if (isDisabled) {
+          return 0; // Return 0 if the item is disabled
+        }
+        return clickedItem;
+      }
+      return undefined;
+    }
+
+     handleKeyDown = (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case Key.ArrowDown:
+          event.preventDefault();
+          this.isUserNavigated = true;
+          // incase of preselected value
+          if (this.activeId === "" && this.value) {
+            this.activeId = this.value;
+          }
+          const currentIndex = this.items.findIndex(item => item.id === this.activeId);
+          if (currentIndex < this.items.length - 1) {
+            this.scrollIndex = currentIndex + 1;
+            this.activeId = this.items[this.scrollIndex].id;  
+          }
+          break;
+
+        case Key.ArrowUp:
+          event.preventDefault();
+          this.isUserNavigated = true;
+          // in case of preselected value
+          if (this.activeId === "" && this.value) {
+            this.activeId = this.value;
+          }
+          const upIndex = this.items.findIndex(item => item.id === this.activeId);
+          if (upIndex > 0) {
+            this.scrollIndex = upIndex - 1;
+            this.activeId = this.items[this.scrollIndex].id;  
+          }
+          break;
+
+        case Key.Enter:
+          event.preventDefault();
+          if (this.activeId) {
+            const selectedItem = this.shadowRoot?.querySelector(`#${prefixId}${this.activeId}`);
+            if (selectedItem) {
+              const isDisabled = selectedItem.getAttribute("aria-disabled") === "true" || selectedItem.hasAttribute("disabled");
+              if (!isDisabled) {
+                this.selectItem(selectedItem as HTMLElement);
+              }
+            }
+          }
+          break;
+
+        default:
+          break;
+      }
+    }
+
+    selectItem(clickedItem: HTMLElement) {
+      if (!clickedItem) return;
+
+      this.activeId = clickedItem.id.substring(clickedItem.id.indexOf("-") + 1);
+      this.selectedItemId = this.activeId;
+      this.updateSelectedState();
+      this.notifySelectedChange();
+    }
+
+    handleClick(event: Event) {
+      event.preventDefault();
+      this.isUserNavigated = false; // Clear navigation flag on click
+      const clickedItem = this.findClickedItem(event);
+      if (clickedItem) {
+        this.scrollIndex = parseInt(clickedItem.getAttribute("index") || "0");
+        this.selectItem(clickedItem);
+      }
+    }
+
+    handleRangeChange = (e: any) => {
+      const { last } = e;
+      this.updateSelectedState();
+      
+      // Trigger 'load-more' event when more items need to be loaded
+      if (this.items.length < this.totalRecords && last >= this.items.length - 1 && !this.isLoading && !this.isError) {
+        this.dispatchEvent(
+          new CustomEvent("load-more", {
+            detail: { page: this.page },
```

Comment:
this load-more event will do the data fetching for new records correct ?
But what if that data loading fails, then how are we taking care of `this.page` in that case, as we are incrementing page here anyways.